src/SSHDebugPS/Docker/DockerConnection.cs

```diff
@@ -242,5 +244,105 @@ private ICommandRunner GetCommandRunner(DockerContainerTransportSettings setting
             else
                 return new RemoteCommandRunner(settings, OuterConnection);
         }
+
+        private const string dockerVersionCommand = "version";
+        private const string dockerVersionArgs = "--format {{.Server.Os}}";
+
+        private const string dockerInfoCommand = "info";
+        private const string dockerInfoArgs = "--format {{.Driver}}";
+
+        public override List<Process> ListProcesses()
+        {
+            string username;
+            TryGetUsername(out username);
```

Comment:
What happens if we fail to get a username?

Reply:
Would also do the same fix for https://github.com/microsoft/MIEngine/blob/4d8d626c7b0ef5150b44bafd9c6475c92b13c9e9/src/SSHDebugPS/PipeConnection.cs#L135

Reply:
In the previous case and in this case we shouldn't care. by returning no username, it should pull all processes. This is the correct behavior.